### ISSUE TITLE
Uncaught TypeError: Cannot read property 'coordinates' of undefined mapbox

### ISSUE THREAD

**Issue by zh0uquan** [alex3165/react-mapbox-gl#218]
So basically in react component `render` method:
if I put a Feature component and a list of Feature component using react wrapper in Layer component, and it will raise `Uncaught TypeError: Cannot read property 'coordinates' of undefined mapbox` even I defined it.
This error costs me hours of debugging, so I guess maybe we can fix this in` Layer` component. 
```
      const NodesFeatures = this.states.nodes.map( (node, i)=> <Feature properties={{rssi: node.rssi}} coordinates={node.location} />);
      <Layer
        id="clusters"
        type="circle"
        paint={{
          "circle-radius": circleRadius,
          "circle-color": {
                property: "rssi",
                stops: [
                    [-300, "#1a237e"],
                    [-120, "#18FFFF"],
                    [-115, "#00E676"],
                    [-110, "#FFEB3B"],
                    [-105, "#FF9800"],
                    [-100, "#FF5722"],
                ]
          },
          "circle-opacity": .8 }}>
        <Feature properties={{rssi: -102}} coordinates={[12.5870103, 52.5170092]} /> // this shoud be removed otherwise it will cause error
        {NodesFeatures} 
      </Layer>
```

**Comment by zh0uquan** [comment]
https://github.com/alex3165/react-mapbox-gl/blob/master/src/layer.ts#232 
maybe we need to change it to `const children = ([] as any).concat(this.props.children)`  to allow children like [element, [list of elements]]???

**Comment by zh0uquan** [comment]
https://github.com/alex3165/react-mapbox-gl/blob/master/src/layer.ts#L232 maybe this line const children = ([] as any).concat(this.props.children || []); this.props.children are type of list like [1, [2,3]]??
and it works like `[].concat([1,[2,3]] || [])` rather than `[].concat(1,[2,3] || [])` as we expected.

**Comment by alex3165** [comment]
Hey @zh0uquan could you reproduce the issue using : https://github.com/alex3165/react-mapbox-gl-debug ? thanks

**Comment by zh0uquan** [comment]
can you check this one? https://github.com/zh0uquan/react-mapbox-gl-debug/tree/master/src

**Comment by zh0uquan** [comment]
https://github.com/alex3165/react-mapbox-gl/blob/master/src/layer.ts#L232
I changed the line to `const children = this.props.children.reduce((arr, el)=> arr.concat(el), [])`, and it works for me

**Comment by alex3165** [comment]
Oh right, I pushed a fix, have a look: https://github.com/alex3165/react-mapbox-gl/commit/d68a4343b81290178fc107c3717ccbae2dff130c can you tell me if it fixed your problem ? Then I will release and publish it.

**Comment by zh0uquan** [comment]
how can I check it? I tried `npm install https://github.com/alex3165/react-mapbox-gl --save`, but instead i got in my `react-mapbox-gl-debug` folder
```
Failed to compile.

Error in ./src/App.js
Module not found: 'react-mapbox-gl' in /root/react-mapbox-gl-debug/src
```

By the way, I was trying to add onClick function to Features inside a **circle type** Layer, but no event is fired, is it supported or I have to hack it?

**Comment by alex3165** [comment]
I published it, try version `1.11.0`

**Comment by zh0uquan** [comment]
Cool, bug fixed. Thank you :)
